makefile

```diff
@@ -36,10 +37,45 @@ $(BIN_DIR) $(OBJECT_DIR):
 clean:
 	@$(RM) -rv $(BIN_DIR) $(OBJECT_DIR)
 
+# ======================================================================== #
+# ============================= Tests Section ============================ #
+# ======================================================================== #
+
+TEST_DIR			:= test
+BIN_TEST_DIR 		:= bin/test
+OBJ_TEST_DIR		:= obj/test
+
+TEST_SOURCES 		:= $(strip $(call rwildcard,$(TEST_DIR),*.c))
+PURE_TEST_SOURCES	:= $(basename $(notdir $(TEST_SOURCES)))
+TEST_OBJECTS 		:= $(patsubst %.c,$(OBJECT_DIR)/%.o,$(notdir $(TEST_SOURCES)))
+TEST_EXES	 		:= $(patsubst %.c,$(BIN_TEST_DIR)/%,$(notdir $(TEST_SOURCES)))
+
+OBJECTS_WITHOUT_MAIN := $(filter-out obj/main.o,$(OBJECTS))
```

Comment:
Forgot to mention that we need to remove the main.o object file because its `main` function symbol clashes with  the test's `main` function when the linker runs.